dataprofiler/data_readers/avro_data.py

```diff
@@ -73,19 +91,19 @@ def is_match(cls, file_path, options=None):
             options = dict()
 
         # get current position of stream
-        if data_utils.is_stream_buffer(file_path):
+        if data_utils.is_stream_buffer(file_path) and not isinstance(file_path, str):
             starting_location = file_path.tell()
 
         is_valid_avro = fastavro.is_avro(file_path)
 
         # return to original position in stream
-        if data_utils.is_stream_buffer(file_path):
+        if data_utils.is_stream_buffer(file_path) and not isinstance(file_path, str):
```

Comment:
why this add?

Reply:
is_stream_buffer checks if file_path is a StringIO or BytesIO but mypy still complained that file_path could still be a string and thus .seek() doesn't exist for strings.

Reply:
Does that mean we have an issue somewhere to fix then?

Reply:
yeah I would look further up the code to see where `file_path` could be a string and try to solve it that way rather than adding the `and not isinstance(file_path, str)`

Reply:
`file_path` is one of the function's parameters and some doctstrings (like the constructor's) says it should be a string. This doesn't seem to be a problem because `fastavro.is_avro(file_path)` accepts `Union[str, IO]`. So the code doesn't seem to be broken but rather mypy isn't playing well with the current way of detecting if `file_path` is a file buffer.

Reply:
We really shouldn't be altering runtime code to make static typing work.

Reply:
outdated diff... this is already in `main`

Reply:
yep somehow got merged in through this PR https://github.com/capitalone/DataProfiler/pull/657/files#diff-da94771d0aad55dc1a9833ead30c7108a37e9eaef5344bae68657c858eab1062R94

Reply:
Can we remove in this PR?

Reply:
I think we should fix / remove in this PR -- whatever the solution may be